docs/csharp/tutorials/mixins-with-default-interface-methods.md

```diff
@@ -0,0 +1,121 @@
+
+In this tutorial, the code doesn't drive IoT devices, but emulates those activities by writing messages to the console. You can explore the code without automating your house.
+
+Next, let's define the interfaces for a light that can automatically turn off after a timeout:
+
+[!code-csharp[pure Timer interface](~/samples/csharp/tutorials/mixins-with-interfaces/UnusedExampleCode.cs?name=SnippetPureTimerInterface)]
+
+You could add a basic implementation to the overhead light, but a better solution is to modify this interface definition to provide a `virtual` default implementation:
+
+[!code-csharp[Timer interface](~/samples/csharp/tutorials/mixins-with-interfaces/ITimerLight.cs?name=SnippetTimerLightFinal)]
+
+By adding that change, the `OverheadLight` can implement the timer function by declaring support for the interface:
+
+```csharp
+public class OverheadLight : ITimerLight
+```
+
+A different light type may support a more sophisticated protocol. It can provide its own implementation fro `TurnOnFor`, as shown in the following code:
+
+[!code-csharp[Override the timer function](~/samples/csharp/tutorials/mixins-with-interfaces/HalogenLight.cs?name=SnippetHalogenLight)]
+
+Unlike overriding virtual class methods, the declaration of `TurnOnFor` in the `HalogenLight` class does not use the `override` keyword. 
+
+## Mix and match capabilities
+
+The advantages of default interface methods become clearer as you introduce more advanced capabilities. Using interfaces enables you to mix and match capabilities. It also enables each class author to choose between the default implementation and a custom implementation. Let's add an interface with a default implementation for a blinking light:
+
+[!code-csharp[Define the blinking light interface](~/samples/csharp/tutorials/mixins-with-interfaces/IBlinkingLight.cs?name=SnippetBlinkingLight)]
+
+The default implementation enables any light to blink. The overhead light can add both timer and blink capabilities using the default implementation:
+
+[!code-csharp[Use the default blink function](~/samples/csharp/tutorials/mixins-with-interfaces/OverheadLight.cs?name=SnippetOverheadLight)]
+
+A new light type, the `LEDLight` supports both the timer function and the blink function directly. This light style implements both the `ITimerLight` and `IBlinkingLight` interfaces, and overrides the `Blink` method:
+
+[!code-csharp[Override the blink function](~/samples/csharp/tutorials/mixins-with-interfaces/LEDLight.cs?name=SnippetOverheadLight)]
+
+An `ExtraFancyLight` might support both blink and timer functions directly:
+
+[!code-csharp[Override the blink and timer function](~/samples/csharp/tutorials/mixins-with-interfaces/ExtraFancyLight.cs?name=SnippetExtraFancyLight)]
+
+The `HalogenLight` you created earlier does not support blinking. You just don't add the `IBlinkingLight` to the list of its supported interfaces.
```

Comment:
thought it could read better by being more direct

```suggestion
The `HalogenLight` you created earlier doesn't support blinking. So, don't add the `IBlinkingLight` to the list of its supported interfaces.
```